### DISCUSSION TITLE
How can I evaluate GCN on NELL dataset?

### DISCUSSION THREAD

**Discussion by chenshih1** [pyg-team/pytorch_geometric#8005] [Q&A]
Hello everyone, i got some problems while evaluating GCN on NELL dataset
the application interface provided by torch_geometric.datasets.NELL encode the feature matrix into coo matrix (sparsetensor), but the x of GCNConv is tensor. meanwhile, functions such as dropout in torch package does not support sparse data either.
i've tried to decode it to dense matrix, but it caused out-of-memory error. so, is there any way to evaluate GCN on NELL dataset using the GCNConv provided by pyg without modifying this framework ?
Thank you !

**Comment by rusty1s** [comment]
The best way is to utilize a `torch.nn.Linear` to map the initial set of features to a smaller space at the beginning of your model. Then, you should have access to the full suite of functionalities from PyG and PyTorch.
